### DISCUSSION TITLE
Errors installing from ISO, never fully starting

### DISCUSSION THREAD

**Discussion by schmug** [Security-Onion-Solutions/securityonion#1708] [Unsupported Versions] [answered]
On a new ISO install, three times, I am getting ERRORs on various services.

![image](https://user-images.githubusercontent.com/38227427/97446049-53b70900-1904-11eb-9c4b-273e077ccbdc.png)

`securityonion/so-strelka-manager                           2.3.2               sha256:ea94ea86dd3d3197c316b24fcd65f85077cdd2fa4e654e260bcbf0fb1a00876f   6e4bb2ca8a38        7 days ago          20.8MB
securityonion-1:5000/securityonion/so-strelka-manager      2.3.2               sha256:ea94ea86dd3d3197c316b24fcd65f85077cdd2fa4e654e260bcbf0fb1a00876f   6e4bb2ca8a38        7 days ago          20.8MB
securityonion-1:5000/securityonion/so-strelka-frontend     2.3.2               sha256:c7a9adeb84f6675d43f13a13c667fa303e9b23ea70bfc178d728660b0f334de5   7e27eee1f26d        7 days ago          21.3MB
securityonion/so-strelka-frontend                          2.3.2               sha256:c7a9adeb84f6675d43f13a13c667fa303e9b23ea70bfc178d728660b0f334de5   7e27eee1f26d        7 days ago          21.3MB
securityonion/so-strelka-filestream                        2.3.2               sha256:3121cee3101343629b2918481d97193b8f9975923607e7a1d3a8118b7186391a   fdc5cfd98a58        8 days ago          20MB
securityonion-1:5000/securityonion/so-strelka-filestream   2.3.2               sha256:3121cee3101343629b2918481d97193b8f9975923607e7a1d3a8118b7186391a   fdc5cfd98a58        8 days ago          20MB
securityonion/so-elastalert                                2.3.2               sha256:37225d521e795e4817b260cd8749d33c1de1955c556acab0192dd5553644bec3   1a5b57a35105        8 days ago          145MB
securityonion-1:5000/securityonion/so-elastalert           2.3.2               sha256:37225d521e795e4817b260cd8749d33c1de1955c556acab0192dd5553644bec3   1a5b57a35105        8 days ago          145MB
securityonion/so-playbook                                  2.3.2               sha256:054049476826732317fe2edce9dd2992ec83cf949f6cff7620cfadc64c8a61b4   c0ad97006462        8 days ago          629MB
securityonion-1:5000/securityonion/so-playbook             2.3.2               sha256:054049476826732317fe2edce9dd2992ec83cf949f6cff7620cfadc64c8a61b4   c0ad97006462        8 days ago          629MB`

I have `docker remove` on each of the ERROR status containers and `so-docker-refresh` and `reboot` to no avail. 

Also of note is mysql shows "unhealthy". Exec'ing into the container and running the commands in the healthy script succeed. 

I chose defaults for all the GUI steps, but setup ended with this: `Errors detected during setup; skipping post-setup steps to allow for analysis of failures.`

```
Error response from daemon: Container daee7a167469a52286a7372319ab065948f7cf6df510b3e4de687531b52941ff is not running
[ERROR   ] Command '['docker', 'exec', 'so-fleet', 'fleetctl', 'get', 'enroll-secret', 'default']' failed with return code: 1
[ERROR   ] stdout: Error response from daemon: Container daee7a167469a52286a7372319ab065948f7cf6df510b3e4de687531b52941ff is not running
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker exec so-fleet fleetctl get enroll-secret default' failed with return code: 1
[ERROR   ] output: Error response from daemon: Container daee7a167469a52286a7372319ab065948f7cf6df510b3e4de687531b52941ff is not running
 </html>nter>nginx/1.17.10</center>ror</h1></center>
[INFO    ] Executing command 'yum check-update 2>&1 | grep "Error: rpmdb open failed"' in directory '/root'
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
```

**Comment by secraven** [comment]
I wish I could help but I have had a number of issues getting this to install correctly.  One thing that helped me was that I needed to let the install sit for a bit before the so-status returned "ok".  I am still struggling with getting the rest of the system to function properly.  I've had SO installed in my home network for years and sure hope I can get this to work!

**Reply by schmug** [reply]
No problem. I am going to nuke it again and try from CentOS7 instead of the ISO and see what happens. I installed 16.04 earlier this year with the ISO and had no troubles.

**Reply by secraven** [reply]
Please keep me posted.

**Reply by schmug** [reply]
CentOS7 minimal and following the docs and I had the same results. Running through the requirements again to see if something changed that could cause my troubles. I can't find what I would consider the main error yet in the logs.

**Comment by TOoSmOotH** [comment]
What kind of hardware are you using?

**Reply by schmug** [reply]
VMware ESXi while evaluating the new version. 12 CPUs, 24 GB ram, 1 TB.
Previous evaluation VM was 16.04 ISO, much smaller specs but same hardware.

**Reply by TOoSmOotH** [reply]
Do a fresh install with the ISO and if it fails let's look at the sosetup.log in /root.

**Reply by schmug** [reply]
At 71% now, do you want the full upload of the log attached here? Thanks for the help.

**Reply by schmug** [reply]
Here's the latest log using the ISO:
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5454679/sosetup.log)

**Reply by secraven** [reply]
I am running a very similar setup (ESXi).  Mine failed during the Post check.   The system eventually reported everything was "ok" from the so-status.  Then I ran into problems accessing Alerts and Hunt.  Elasticsearch wasn't working either.

**Reply by secraven** [reply]
GOOD NEWS!  I got it working.  I didn't an install using Centos 7.  The difference is that I chose Evaluation vs. Production.  I also only chose Grafana & Strelka.  For the older SO I would choose production.

**Reply by schmug** [reply]
I will try that tomorrow, I thought I had tried Standalone (Production?) and Evaluation on one of my installs but I cannot remember. Are you saying you used Centos 7 or the ISO with evaluation mode?

Since I had plans to go to production eventually I wanted to start with Standalone, and then Distributed after it worked as expected depending on resource needs.

**Reply by secraven** [reply]
I did a minimal CentOS 7 install and ran the installer.

Question:
- Is it possible to add OSQUERY after the installation has been completed?

**Reply by schmug** [reply]
I tried the ISO using Evaluation and it seemed to be the same issues.

**Reply by secraven** [reply]
I did the ISO install too using Evaluation and it worked.  I chose not to install The Hive.  The so-status had some errors but after a while they went away.  I needed to restart mysql.  I am now testing the forwarding of syslog from my firewall to this install.

**Reply by schmug** [reply]
I tried a Distributed install from ISO, and I chose Manager with everything else deselected. Still getting so-elastalert in ERROR state and can't access the web interface after so-allow.

**Comment by Ishfal** [comment]
I have same issue on preexisting 2.3.1 installation that I upgraded to 2.3.2. Refreshing the Docker images as per above resolved all of them except for so-elastalert, which remains in 'Error' state. so-mysql also shoes as being in 'unhealthy' state though otherwise seems to be operating normally.

**Comment by schmug** [comment]
I've tried 2.2.0-rc3.iso, CentOS 7 with git, and Ubuntu 18.04 and I'm not getting any further. 

I had someone from another company try as well and could not get the current ISO working with ESXi (could be a common thread?), legacy boot.

From the recent Ubuntu install:
```
Enabling Fleet...
[ERROR   ] Command '['docker', 'exec', 'so-fleet', 'fleetctl', 'get', 'enroll-secret', 'default']' failed with return code: 1
[ERROR   ] stdout: Error response from daemon: Container 090890eee4592ce58634926f5d5a51f33ac8615fecfb97b608fc1f6a5dfbbc11 is not running
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker exec so-fleet fleetctl get enroll-secret default' failed with return code: 1
[ERROR   ] output: Error response from daemon: Container 090890eee4592ce58634926f5d5a51f33ac8615fecfb97b608fc1f6a5dfbbc11 is not running
[sosetup-ubuntu.log](https://github.com/Security-Onion-Solutions/securityonion/files/5487693/sosetup-ubuntu.log)

[CRITICAL] Rendering SLS 'base:fleet.event_enable-fleet' failed: mapping values are not allowed in this context
```

The Ubuntu install took massively longer than any other method, hours.

[sosetup-ubuntu.log](https://github.com/Security-Onion-Solutions/securityonion/files/5487699/sosetup-ubuntu.log)

**Comment by schmug** [comment]
I tried UEFI with the ISO as well, no luck and the same result. I've gone back to the 16.04.7.1 ISO and all services are running properly.

Standalone/Production/Best Practices with all of the same hardware.

**Comment by Gunbird** [comment]
I hit the same errors during install of SO 2.3.2 from iso on a kvm VM.

Setup ends with `Errors detected during setup; skipping post-setup steps to allow for analysis of failures.`

I installed in eval mode.

I did the install 3 times with the same result.

I have the exact same services in error state as schmug.

docker refresh didn't help.

The sosetup.log does not give very explicit messages about the cause of the errors.

Errors in sosetup.log :

```
[INFO    ] {'image': {'Time_Elapsed': 0.03319573402404785, 'retcode': 0, 'Layers': {'Already_Pulled': ['75f829a71a1c'], 'Pulled': ['b9f272a0e2df', 'dce5d1796f35', '3042827e1add', '27675ba9d981', 'c11476d187eb', '3f3cc16bd4a3', '520846a7c
80b', '55e74ce8834c', 'b1cbda46ac36', 'cf547d31a6a1', 'd8f3e1c0686a', 'e8c978046529', '66aa6e8cc36c', '2d65b25c6bd4', '24badd811f14', 'c1612d55896c', 'a4d30842e094', '7603f08f1b37', '50bf8f4f4524']}, 'Status': 'Downloaded newer image for
 bermudes:5000/securityonion/so-kibana:2.3.2'}, 'container_id': {'added': '80febf638bf21f28e93970966a33c40758a4939814c9906a98d44e8bf09ce560'}, 'state': {'old': None, 'new': 'running'}}
[INFO    ] Completed state [so-kibana] at time 20:44:22.650206 (duration_in_ms=34000.646)
[INFO    ] Running state [/opt/so/conf/kibana/saved_objects.ndjson.template] at time 20:44:22.650666
[INFO    ] Executing state file.managed for [/opt/so/conf/kibana/saved_objects.ndjson.template]
[INFO    ] Fetching file from saltenv 'base', \*\* done \*\* 'kibana/files/saved_objects.ndjson'
[INFO    ] File changed:
New file
[INFO    ] Completed state [/opt/so/conf/kibana/saved_objects.ndjson.template] at time 20:44:22.757828 (duration_in_ms=107.16)
[INFO    ] Running state [wait_for_kibana] at time 20:44:22.761455
[INFO    ] Executing state module.run for [wait_for_kibana]
[INFO    ] {'http.wait_for_successful_query': {'status': 503, 'error': 'HTTP 503: Service Unavailable'}}
[INFO    ] Completed state [wait_for_kibana] at time 20:47:22.766562 (duration_in_ms=180005.107)
[INFO    ] Running state [/usr/sbin/so-kibana-config-load] at time 20:47:22.768138
[INFO    ] Executing state cmd.run for [/usr/sbin/so-kibana-config-load]
[INFO    ] Executing command '/usr/sbin/so-kibana-config-load' in directory '/opt/so'
[INFO    ] {'pid': 9243, 'retcode': 0, 'stdout': '', 'stderr': ''}
[INFO    ] Completed state [/usr/sbin/so-kibana-config-load] at time 20:47:22.818735 (duration_in_ms=50.597)
local:
----------
          ID: kibanasearchgroup
    Function: group.present
```

```
         ID: wait_for_kibana
    Function: module.run
      Result: True
     Comment: http.wait_for_successful_query: Success
     Started: 20:44:22.761455
    Duration: 180005.107 ms
     Changes:   
              ----------
              http.wait_for_successful_query:
                  ----------
                  error:
                      HTTP 503: Service Unavailable
                  status:
                      503
----------
          ID: so-kibana-config-load
    Function: cmd.run
        Name: /usr/sbin/so-kibana-config-load
      Result: True
     Comment: Command "/usr/sbin/so-kibana-config-load" run
     Started: 20:47:22.768138
```
```

[INFO    ] Completed state [/usr/sbin/so-playbook-ruleupdate > /opt/so/log/playbook/update.log 2>&1] at time 20:48:16.238727 (duration_in_ms=10.535)
[INFO    ] Running state [create_user] at time 20:48:16.240182
[INFO    ] Executing state cmd.script for [create_user]
[INFO    ] Fetching file from saltenv 'base', \*\* done \*\* 'playbook/files/automation_user_create.sh'
[INFO    ] Executing command '/root/__salt.tmp.c4karwa5.sh' in directory '/root'
[INFO    ] {'pid': 11100, 'retcode': 0, 'stdout': '', 'stderr': ''}
[INFO    ] Completed state [create_user] at time 20:48:26.415937 (duration_in_ms=10175.751)
[INFO    ] Running state [until nc -z 172.17.211.2 3200; do sleep 1; done] at time 20:48:26.416427
[INFO    ] Executing state cmd.run for [until nc -z 172.17.211.2 3200; do sleep 1; done]
[INFO    ] Executing command 'until nc -z 172.17.211.2 3200; do sleep 1; done' in directory '/root'
[ERROR   ] {'stdout': 'until nc -z 172.17.211.2 3200; do sleep 1; done : Timed out after 30 seconds', 'stderr': '', 'retcode': 1, 'pid': 11332}
[INFO    ] Completed state [until nc -z 172.17.211.2 3200; do sleep 1; done] at time 20:48:56.426989 (duration_in_ms=30010.562)
local:
----------
          ID: mysqlpkgs
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 20:48:11.349830
    Duration: 776.659 ms
     Changes:
```

```
Failed:    0
------------
Total states run:     5
Total run time:   3.037 s
----
76% - APPLYING SO-FLEET-SETUP SALT STATE
----
[+] Set the address config key to "https://localhost:8080" in the "default" context
[+] Set the tls-skip-verify config key to "true" in the "default" context
[+] Set the url-prefix config key to "/fleet" in the "default" context
Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
Enabling Fleet...
[ERROR   ] Command '['docker', 'exec', 'so-fleet', 'fleetctl', 'get', 'enroll-secret', 'default']' failed with return code: 1
[ERROR   ] stdout: Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker exec so-fleet fleetctl get enroll-secret default' failed with return code: 1
[ERROR   ] output: Error response from daemon: Container 605896fc9f1afc1889b0347bffca267f2bf568121dec8b9f4712f751ba683219 is not running
[CRITICAL] Rendering SLS 'base:fleet.event_enable-fleet' failed: mapping values are not allowed in this context
Generating osquery install packages - this will take some time...
Installing launcher via salt...
so-nginx
Fleet Setup Complete - Login with the username and password you ran the script with.
----
78% - APPLYING WAZUH SALT STATE
----
[INFO    ] Loading fresh modules for state activity
[INFO    ] Fetching file from saltenv 'base', \*\* done \*\* 'wazuh/init.sls'
```

```
... Setting ONBOOT for management interface
... Copying 99-so-checksum-offload-disable
... Modifying 99-so-checksum-offload-disable
----
87% - ADDING USER TO SOC
----
File /opt/so/conf/kratos/db/db.sqlite found at Sun Nov 15 21:02:44 UTC 2020
Attempting to add administrator user for web interface...
Successfully added new user to SOC
Unable to add user to TheHive; user might already exist
<html>^M <head><title>500 Internal Server Error</title></head>^M <body>^M <center><h1>500 Internal Server Error</h1></center>^M <hr><center>nginx/1.17.10</center>^M </body>^M </html>
Add user result: 0
----
90% - ENABLING CHECKIN AT BOOT
----
Enabling checkin at boot
----
95% - VERIFYING SETUP
----
[INFO    ] Loading fresh modules for state activity
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/clouds'
```

And finally : 
```
   Duration: 77.129 ms
     Changes:   

Summary for local
--------------
Succeeded: 439 (changed=52)
Failed:      0
--------------
Total states run:     439
Total run time:   175.974 s
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
Installer removing the following files:
/root/installtmp:
total 0
drwxr-xr-x. 3 root root 21 Nov 15 20:35 pillar

/root/installtmp/pillar:
total 0
drwxr-xr-x. 2 root root 31 Nov 15 20:37 minions

/root/installtmp/pillar/minions
```

Thanks for your help

**Comment by schmug** [comment, marked as answer]
With 2.3.10, I've had a distributed install succeed.

**Reply by Gunbird** [reply]
I confirm that 2.3.10 fixed this issue for me too.
